### PR TITLE
Implement ClientConnectEvent

### PR DESCRIPTION
**Pull request by simon816**
Implementation of https://github.com/SpongePowered/SpongeAPI/pull/663

Tested and works as expected

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.interfaces.IMixinNetHandlerLoginServer;
+
+@Mixin(targets = "net/minecraft/server/network/NetHandlerLoginServer$2")
+public class MixinNetHandlerLoginServerAnonThread extends Thread {
+
+    @Shadow private NetHandlerLoginServer this$0;
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: Where can I find the SRG name for this synthetic field? @Mumfrey?

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: I'm pretty sure you need to find the obfuscated class name, then look it up by the field name `a`. `a` is always mapped to the first field, and `this$0` is always the first field in the class, at least according to `javac`.

Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: @JBYoshi Didn't work unfortunately.
The notch name for `NetHandlerLoginServer$2` is `rs`, but the field `rs.a` doesn't exist

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: It should be `field_151286_b`. Look up the field in the notch/srg mappings.

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: Okay, not so sure about that. Further searching makes it look like it's supposed to be `field_180221_a`.

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: I confused `NetHandlerLoginServer` and `NetHandlerPlayServer`. It is `field_180221_a`.

Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/server/network/MixinNetHandlerLoginServerAnonThread.java]: Yep that works. It's interesting that that field does not appear in MCPBot searches.

**Pull request merged**